protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java

```diff
@@ -216,6 +223,23 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
         return serviceStatus;
     }
 
+    private void setFinalStatic(final Field finalStaticField, final Object value) throws NoSuchFieldException, IllegalAccessException {
+        finalStaticField.setAccessible(true);
+
+        final Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(finalStaticField, finalStaticField.getModifiers() & ~Modifier.FINAL);
+
+        finalStaticField.set(null, value);
+    }
+
+    private void setJCifsTimeouts(int t) throws NoSuchFieldException, ClassNotFoundException, IllegalAccessException {
```

Comment:
This doesn't appear to be thread safe. If we have two monitors with different timeouts executing at the same time, which timeout will they get?

Reply:
There doesn't appear to be an easy way of working around this given the way JCifs is structured. Let's just add a note about this limitation in the docs and in the code.